apollo-router/src/compute_job.rs

```diff
@@ -14,13 +14,25 @@ use crate::metrics::meter_provider;
 /// reaches `QUEUE_SOFT_CAPACITY_PER_THREAD * thread_pool_size()`
 const QUEUE_SOFT_CAPACITY_PER_THREAD: usize = 20;
 
-/// Let this thread pool use all available resources if it can.
+/// By default, let this thread pool use all available resources if it can.
 /// In the worst case, we’ll have moderate context switching cost
 /// as the kernel’s scheduler distributes time to it or Tokio or other threads.
 fn thread_pool_size() -> usize {
-    std::thread::available_parallelism()
-        .expect("available_parallelism() failed")
-        .get()
+    // This environment variable is intentionally undocumented.
+    // See also APOLLO_ROUTER_IO_THREADS in apollo-router/src/executable.rs
+    if let Some(size) = std::env::var("APOLLO_ROUTER_COMPUTE_THREADS")
+        .ok()
+        .and_then(|value| value.parse::<usize>().ok())
+    {
+        tracing::debug!("Compute thread pool size: $APOLLO_ROUTER_COMPUTE_THREADS = {size}");
+        size
+    } else {
+        let size = std::thread::available_parallelism()
+            .expect("available_parallelism() failed")
+            .get();
+        tracing::debug!("Compute thread pool size: available_parallelism() = {size}");
```

Comment:
This log can also help makes sure cgroup limits affect `available_parallelism()` like we expect when benchmarking

Reply:
```
tracing::info!(size, type="compute", source="calculated", "thread pool");
```